panic: runtime error: slice bounds out of range [:-1] again

### Nuclei version: 2.9.2-dev latest commit [bca0c53a41e2d8cd58779369eb948dacfec360d7](https://github.com/projectdiscovery/nuclei/commit/bca0c53a41e2d8cd58779369eb948dacfec360d7)
`

### Current Behavior:
crashing while scanning a large list

### Expected Behavior:
Save resume file so if crashes like in this case we can resume from last saved point

### Steps To Reproduce:
`cat ips.txt | nuclei -t cves -t exposures -t default-logins -t vulnerabilities -t exposed-panels -t file -stats -rl 150-o ips.nuclei
`
### Anything else:
fails somewhere parsing client certificate

```
panic: runtime error: slice bounds out of range [:-1]

goroutine 24309671 [running]:
github.com/projectdiscovery/tlsx/pkg/tlsx/clients.FormatToSerialNumber(0xc02087dc0a?)
        /root/go/pkg/mod/github.com/projectdiscovery/tlsx@v1.0.6/pkg/tlsx/clients/utils.go:103 +0x189
github.com/projectdiscovery/tlsx/pkg/tlsx/clients.Convertx509toResponse(0xc032526828?, {0xc0225cfd28, 0xd}, 0xc02e8c5180, 0x0)
        /root/go/pkg/mod/github.com/projectdiscovery/tlsx@v1.0.6/pkg/tlsx/clients/utils.go:35 +0x37a
github.com/projectdiscovery/tlsx/pkg/tlsx/tls.(*Client).ConnectWithOptions(0xc005006f78, {0xc0225cfd28, 0xd}, {0xc0225cfd28, 0xd}, {0xc0225cfd36, 0x3}, {{0x0, 0x0}, {0x0, ...}, ...})
        /root/go/pkg/mod/github.com/projectdiscovery/tlsx@v1.0.6/pkg/tlsx/tls/tls.go:153 +0x7ff
github.com/projectdiscovery/tlsx/pkg/tlsx/auto.(*Client).ConnectWithOptions(0xc005006fc0, {0xc0225cfd28, 0xd}, {0xc0225cfd28, 0xd}, {0xc0225cfd36, 0x3}, {{0x0, 0x0}, {0x0, ...}, ...})
        /root/go/pkg/mod/github.com/projectdiscovery/tlsx@v1.0.6/pkg/tlsx/auto/auto.go:42 +0xb8
github.com/projectdiscovery/tlsx/pkg/tlsx.(*Service).ConnectWithOptions(0xc005006f60, {0xc0225cfd28, 0xd}, {0xc0225cfd28, 0xd}, {0xc0225cfd36, 0x3}, {{0x0, 0x0}, {0x0, ...}, ...})
        /root/go/pkg/mod/github.com/projectdiscovery/tlsx@v1.0.6/pkg/tlsx/tlsx.go:74 +0x243
github.com/projectdiscovery/tlsx/pkg/tlsx.(*Service).Connect(...)
        /root/go/pkg/mod/github.com/projectdiscovery/tlsx@v1.0.6/pkg/tlsx/tlsx.go:60
github.com/projectdiscovery/nuclei/v2/pkg/protocols/ssl.(*Request).ExecuteWithResults(0xc0009f49a0, 0xc0378a6440, 0x7f8859ea8df8?, 0x7f88830d1a68?, 0xc044fdca80)
        /root/go/pkg/mod/github.com/projectdiscovery/nuclei/v2@v2.9.1-0.20230413124104-bca0c53a41e2/pkg/protocols/ssl/ssl.go:218 +0x62d
github.com/projectdiscovery/nuclei/v2/pkg/templates.(*ClusterExecuter).Execute(0xc002165140, 0xc0378a6380)
        /root/go/pkg/mod/github.com/projectdiscovery/nuclei/v2@v2.9.1-0.20230413124104-bca0c53a41e2/pkg/templates/cluster.go:245 +0x22d
github.com/projectdiscovery/nuclei/v2/pkg/core.(*Engine).executeTemplateWithTargets.func2.1(0x7566a6?, 0x0?, 0xc005931920)
        /root/go/pkg/mod/github.com/projectdiscovery/nuclei/v2@v2.9.1-0.20230413124104-bca0c53a41e2/pkg/core/executors.go:128 +0x296
created by github.com/projectdiscovery/nuclei/v2/pkg/core.(*Engine).executeTemplateWithTargets.func2
        /root/go/pkg/mod/github.com/projectdiscovery/nuclei/v2@v2.9.1-0.20230413124104-bca0c53a41e2/pkg/core/executors.go:105 +0x4f8
```

--- a/v2/go.mod
+++ b/v2/go.mod
@@ -78,7 +78,7 @@ require (
 	github.com/projectdiscovery/ratelimit v0.0.6
 	github.com/projectdiscovery/rdap v0.9.1-0.20221108103045-9865884d1917
 	github.com/projectdiscovery/sarif v0.0.1
-	github.com/projectdiscovery/tlsx v1.0.6
+	github.com/projectdiscovery/tlsx v1.0.7
 	github.com/projectdiscovery/uncover v1.0.2
 	github.com/projectdiscovery/utils v0.0.20-0.20230410133604-010edb62cb35
 	github.com/projectdiscovery/wappalyzergo v0.0.88

